### ISSUE TITLE
Legacy mode

### ISSUE THREAD

**Issue by mansellan** [WaynePhillipsEA/twinbasic#65]
**Is your feature request related to a problem?**
It should be possible to specify that a given twinBASIC project only allows legacy syntax.

**Describe the solution you'd like**
A project-level switch that disallows any syntax that would not be valid in legacy VB6.

**Additional context**
twinBASIC includes superb language extensions, for example generics and overloads. However, use of these renders source code incompatible with the legacy VB6 compiler. It should be possible to specify a project as "legacy mode", whereby the compiler only allows legacy-compatible syntax.

**Comment by bclothier** [comment]
FWIW, I wouldn't want it to be only a project-level switch. Imagine that I've inherited a old codebase and I am asked to add new feature. I probably want to enable it as I go through the codebase, maybe on a per-file level or per-namespace level, allowing me to amortize the tech debt payments.

That is a slightly different scenario than supporting compatibility with VB6 users but something to consider.

**Comment by WaynePhillipsEA** [comment]
I agree, this should probably be an option per file/component as well, much like how Option Explicit is currently handled.

**Comment by Kr00l** [comment]
Will this mode be an on - off switch, or a drop-down choice?
I would imagine a drop-down as following..
Legacy mode:
- none (default, full feature set)
- VB6/VBA6
- VBA7 (allowing LongLong, LongPtr, 64 bit etc.)

**Comment by Kr00l** [comment]
When legacy mode is on, will then #61 do default to HKLM instead of HKCU?

**Comment by WaynePhillipsEA** [comment]
@Kr00l Yes, I expect we'll have to do that.

**Comment by mansellan** [comment]
Just had a wacky idea. It may be nuts, hence not opening a new issue just yet...

How about a project option, available only in legacy-mode projects, for users who have VB6 installed, to send compilation to the actual VB6 compiler?

It would be a strange thing to do, for sure. But if someone is wedded to the legacy compiler (perhaps they want p-code?), they could get all the modern conveniences of VSCode but still compile exactly as before. Of course, it wouldn't be debuggable in VSC from that form, so probably only of use as a final stage after debugging with tB.

Am I insane?

**Comment by mansellan** [comment]
> 
> 
> > Am I insane?
> 
> Yes, yes you are. 😜

Good, thanks for confirming :-)

**Comment by WaynePhillipsEA** [comment]
Don't worry, you're in good company here.

**Comment by DaveInCaz** [comment]
One use case is editing a shared file that needs to retain compile-ability both in VB6 and also in tB. This couldn't be done with a file-level `Option Legacy` because (ironically) that would not be backwards compatible. It would have to be project level or something else.

**Comment by bclothier** [comment]
Just to float a possible suggestion.... is it easier to solve if it was made as a build target? We have win32 and win64; if we then create a new build target, let's call it "vb6" that would follow the VB6's grammar strictly and treat any VBA7 and tB extensions as incompatible. 

By using build configuration instead of a project configuration, we can make it easier to make explicit the intended audience as discussed in #16 and hopefully encourage a better separation. If I'm upgrading a large project, I probably will want to have my new code in a new project that the original project can reference safely via compatible interfaces.

**Comment by mwolfe02** [comment]
> is it easier to solve if it was made as a build target?

Interesting thought.  Especially as it relates to package management.  Is part of this idea that it would better support a potential ecosystem of tB packages that target VB6?  In some ways, this could become a de facto VB6 package manager, as nothing like that really exists now (correct?).
